clcache.py

```diff
@@ -650,6 +650,13 @@ def expandBasedirPlaceholder(path, baseDir):
         return path
 
 
+def collapseBasedirToPlaceholder(path, baseDir):
```

Comment:
How about an

```python
assert path == os.path.normcase(path)
```

I tend to prefer documenting requirements either in types (but Python doesn't have a terribly strong type system, in particular it has no `Path` class or the like AFAIK) or at least `assert`ions which are enforced at runtime.